lib/src/revset.rs

```diff
@@ -1630,6 +1657,7 @@ impl<'a> DefaultSymbolResolver<'a> {
                 .iter()
                 .flat_map(|ext| ext.as_ref().new_resolvers(repo))
                 .collect(),
+            workspace_ctx,
```

Comment:
`HEAD@git` would have to be translated to `CommitRef::GitHead(WorkspaceId)` or something at `lower_expression()` (who knows the current workspace.)

Reply:
BTW, it might be okay to deprecate `HEAD@git` symbol since it can no longer be abstracted as a remote ref of the local `@git` remote. It serves as an indicator of colocated repo, but I've never used the `HEAD@git` symbol, which should usually point to `@-`.

Reply:
I had to add the workspace id to the resolver anyway, because the resolver needs to be able to generate HEAD@git as a did-you-mean suggestion, and must do that only when the current workspace is colocated. Since it had to be added to the resolver anyway, it was simple to resolve `GitHead` without changing CommitRef.

Reply:
Hmm, good point. I lean towards deprecating `HEAD@git` then.

As I said, it's no longer a valid abstraction, and the HEAD revision can be queried by `git_head()`. IIRC, `HEAD@git` resolution was added just because `git_head` template is rendered as such, not because we have a real use case.

Reply:
https://github.com/martinvonz/jj/pull/4632